homeassistant/helpers/storage.py

```diff
@@ -286,16 +288,36 @@ def async_delay_save(
             "data_func": data_func,
         }
 
+        now = self.hass.loop.time()
+
+        if (
```

Comment:
An alternative approach could be that if a timer handle is active, we store a "next timestamp" at which we actually want to be called. When the timer handle fires and it sees there is a "next timestamp", it schedules a new timer handle for "next timestamp" instead of doing the work.

Reply:
That's a good idea. It's the cancellation of the timer handle that is expensive.  If they fire it's cheap to remove because it's on the top of the heap

Reply:
implemented as suggested 👍  Its cleaner